airbyte-cdk/python/unit_tests/sources/declarative/incremental/test_datetime_based_cursor.py

```diff
@@ -524,6 +524,48 @@ def test_parse_date(test_name, input_date, date_format, date_format_granularity,
     assert expected_output_date == output_date
 
 
+@pytest.mark.parametrize(
+    "test_name, input_date, date_formats, expected_output_date",
+    [
```

Comment:
Should we have a test case that we use `datetime_format` if `date_formats` is empty?

Reply:
I kept the previous tests which are doing that